### PR TITLE
feat(Routing): change to declerative routing

### PR DESCRIPTION
**Pull request by datner**
This is a suggestion, will be nice to have.
Logic is untouched

### CODE REVIEW

File: src/routes/Raid.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const userStatSorts: {
 }
 const userStatSortNames = Object.keys(userStatSorts)
 
-const ViewRaid = () => {
+const Raid = () => {
```
Comment (JacobMGEvans) [inline src/routes/Raid.tsx:20]: Why the rename?

File: src/routes/Raids.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from 'react'
 
-import { Link } from 'react-router-dom'
+import { Link, useRouteMatch } from 'react-router-dom'
 import LoadingSpinner from '#components/loading-spinner'
 import useCollection from '#utils/useCollection'
 
-const AllRaids = () => {
+const Raids = () => {
```
Comment (JacobMGEvans) [inline src/routes/Raids.tsx:7]: Why the change in the naming convention?

File: src/App.tsx [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+  <React.Suspense fallback={<LoadingSpinner />}>
+    <Switch>
+      <Route path="/raids" exact component={Raids} />
+      <Route path="/raids/:raidId" exact component={Raid} />
+      <Route path="/" exact component={Home} />
+      <Route path="*" component={NotFound} />
+    </Switch>
+  </React.Suspense>
+)
+
```
Comment (JacobMGEvans) [inline src/App.tsx:24]: 👍🏻 I like this mainly because I prefer the explicitness

File: src/routes/Raid.tsx (hunk 1 above)
Comment (datner) [reply src/routes/Raid.tsx:20]: Just wanted to perserve the Singular-Plural theme, I can change it back

File: src/routes/Raids.tsx [hunk 4]
```diff
@@ -30,25 +31,25 @@ const AllRaids = () => {
...
+
+  return collectionData.state === 'loading' ? (
+    <LoadingSpinner />
+  ) : (
+    // Theoretically this can/should never be hit... But a fun message nonetheless
+    <p>
+      {`Strange... I could've sworn the Manticore was here a second ago! Seems
+        there aren't any Raids right now. Try again later!`}
+    </p>
+  )
```
Comment (JacobMGEvans) [inline src/routes/Raids.tsx:52]: Makes sense the logic would get hit regardless of `else` block or not 👍🏻

File: src/routes/Raid.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply src/routes/Raid.tsx:20]: Cool, just want to make sure there aren't arbitrary "personal preference" changes or getting into "bikeshedding". 

I am always down for discourse and what not though.

File: src/routes/Raids.tsx (hunk 2 above)
Comment (datner) [reply src/routes/Raids.tsx:7]: i borrowed from REST, again, if this is out of line I don't mind changing it back

File: src/App.tsx (hunk 3 above)
Comment (datner) [reply src/App.tsx:24]: same <3

File: src/routes/Raids.tsx (hunk 2 above)
Comment (JacobMGEvans) [reply src/routes/Raids.tsx:7]: Nothing is really "out of line" I want to continue to perpetuate crowdsourcing and mostly mobocracy.

Comment (datner) [reply src/routes/Raids.tsx:7]: Then this is my proposal 😄 
Open to feedback

**Comment by JacobMGEvans** [comment]
LGTM :tada: 
Gonna have Braydon take a look since he worked on this stuff

**Comment by datner** [comment]
sure thing

File: src/routes/Raid.tsx (hunk 1 above)
Comment (nobrayner) [reply src/routes/Raid.tsx:20]: I was trying to void almost identical names - especially after the experience with HospitalRun... but if there are good reasons to avoid this, then I'm all ears!

**Review by nobrayner — COMMENTED** [review]
The only thing I'm not a fan of is using `Raid` and `Raids`... I've found having a difference of one letter - while it may be "more correct" - just makes life annoyingly confusing when trying to jump to files.

**Comment by datner** [comment]
So should I change back or maybe change to something else? I don't want the PR to fall on this

**Comment by JacobMGEvans** [comment]
> So should I change back or maybe change to something else? I don't want the PR to fall on this

Yes change it back please

**Comment by datner** [comment]
done

**Review by JacobMGEvans — APPROVED** [review]
